src/parse.ts

```diff
@@ -55,6 +55,39 @@ export function parseMessage(
   return output as ParsedMessage;
 }
 
+export function parseTimestamp(
+  definition: MessageDefinition,
+  definitions: Map<string, MessageDefinition>,
+  view: DataView,
+  offset = 0,
+): bigint {
+  let timestamp: bigint | undefined;
+
+  let curOffset = offset;
+  for (const field of definition.fields) {
+    if (field.name.startsWith("_")) {
+      continue;
+    }
+    if (field.name === "timestamp") {
```

Comment:
feels like this loop could be simplified using `find()` instead of a loop.

However, I'm also noticing that you will do this loop through fields on each message. If you are only trying to parse timestamps for all messages, it would be faster to find the timestamp field once and skip the linear search through `definition.fields` on each message. that is - hoist the find() outside of this parseTimestamp function and pass the field in.

Reply:
hm - but different messages have different definition fields...

Reply:
Is there some “topic”/“schema” concept we can tie it to?